prepare-image.sh

```diff
@@ -8,7 +8,7 @@ echo "install_weak_deps=False" >> /etc/dnf/dnf.conf
 # Tell RPM to skip installing documentation
 echo "tsflags=nodocs" >> /etc/dnf/dnf.conf
 
-dnf install -y python3 python3-requests 'dnf-command(config-manager)'
+dnf install -y python3 python3-requests 'dnf-command(config-manager)' gcc make git-core perl xz-devel
```

Comment:
Changes to this file might be a topic of debate or topic of deeper discussion because this affects the size of the OCI image.

But I would really like to support this functionality so your input is very welcome @elfosardo @dtantsur

Reply:
As discussed on the meeting, could you please consider creating a new container image (akin to ipa-downloader) to build the iPXE firmware? These new dependencies are quite heavy, and in any case we should avoid piling more functionality to ironic-image that is not about launching ironic or something absolutely required for it.

Reply:
+1 as we discussed in the meeting if this is going to be an optional initContainer IMO it would be better to create a new container image, particularly given the additional dependencies aren't particularly small

Reply:
- I have no objections to a separate container image, your arguments make sense to me.

- What do you think should the separate container live in this repo, or should it have it's own repo?

- Adding an additional repo that has to be versioned, branched, tested separately and it would be considerable work for this relatively small image. But on the other hand ipa-downloader is even simpler and smaller and it has it's own repo so I am not against that either. @tuminoid @kashifest @lentzi90  your input would be appreciated also.

Reply:
Separate container image sounds good to me. Separate repositories for every image sounds a bit much to me though. I would prefer to merge at least some of them. Something like a "ironic-image-utils" repo?

Reply:
+1 to @lentzi90's proposal. I also don't see a need for separate repository, specially since we are already struggling to manage versioning/branching etc for existing images like ironic-image, mariadb-image or do we already have clear versioning schemes in mind for all these exisinting repos and those that we propose to come in future?

Reply:
Okay, thanks for the responses, I was also thinking about this, and I would propose to have this ipxe builder as part of the ironic-image repository as a separate image but also I would like to bring the IPA-downloader and the ironic-client images under the same roof in the ironic-image repository. I believe it would be beneficial if all this component/utility images would be versioned together and live on the same release branches together with the ironic-imagein the future (when we will have those) as they are all related in some way.

Reply:
Yes, I agree with @Rozzii  latest comment. We should group these so they're more manageable.

Reply:
+1 I agree it doesn't necessarily need to be a new repository provided we can build the images independently - we do already have some utility images in this repo in the [resources subdir](https://github.com/metal3-io/ironic-image/tree/main/resources) although those are only test related, not runtime dependencies